spec/services/update_auction_spec.rb

```diff
@@ -2,6 +2,43 @@
 
 describe UpdateAuction do
   describe '#perform' do
+    context 'when publishing an auction' do
+      it 'creates an AuctionState record' do
+        auction = create(:auction, :unpublished)
+
+        params = {
+          auction: {"published" => "published"}
+        }
+
+        expect do
+          UpdateAuction.new(
+            auction: auction,
+            params: params,
+            current_user: auction.user
+          ).perform
+        end.to change { AuctionState.count }.by(1)
+
+        auction.reload
+
+        # do we want to low-level test for the published state object, like this?
+        published_state = auction.states.find {|state| state.name == 'published'}
+        expect(published_state.state_value).to eq('published')
+
+        # and/or this?:
+        # expect(auction.published?).to be true
```

Comment:
This doesn't seem important to me. It seems like that is the interface expansion that I was complaining about. Special methods for states instead of just setting state and reading state. Since the states are distributed now, *we will need a nice way to read states.* `auction.state_for('published') # "archived" ` ???